CompilerSource/ipc/Sprite.fbs

```diff
@@ -0,0 +1,39 @@
+include "Point.fbs";
+include "Image.fbs";
+
+enum SpriteShape : byte { PRECISE, RECTANGLE, DISK, DIAMOND }
+
+enum SpriteBoundingBoxType : byte { AUTOMATIC, FULL_IMAGE, MANUAL }

```

Comment:
Is SpriteBoundingBoxType a little long for yah? I prefer naming all of this "Type" over "Kind" because it's just less awkward. Kind is more often used to link something as belonging to a group, not to name the group itself. I don't know which is more semantically correct. Kind is also more informal than Type. Thus it seems it would be more appropriate to postfix "Kind" to the field name and "Type" to field type?